src/main/java/service/WalletService.java

```diff
@@ -0,0 +1,86 @@
+package service;
+
+import entity.CryptoCurrency;
+import entity.Transaction;
+import entity.User;
+import entity.Wallet;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.dao.EmptyResultDataAccessException;
+import repository.CryptoCurrencyRepository;
+import repository.UserRepository;
+import repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+public class WalletService {
+
+    private final WalletRepository walletRepository;
+    private final CryptoCurrencyRepository cryptoCurrencyRepository;
+    private final UserRepository userRepository;
+
+    @Autowired
+    public WalletService(WalletRepository walletRepository, CryptoCurrencyRepository cryptoCurrencyRepository, UserRepository userRepository) {
+        this.walletRepository = walletRepository;
+        this.cryptoCurrencyRepository = cryptoCurrencyRepository;
+        this.userRepository = userRepository;
+    }
+
+    public Wallet createWalletWithCurrencies(BigDecimal balance, String currency, Long userId, List<Long> cryptoCurrencyIds) {
+        User existingUser = userRepository.findById(userId)
+                .orElseThrow(() -> new EmptyResultDataAccessException("User not found with ID: " + userId, 1));
+
+        List<CryptoCurrency> cryptoCurrencies = cryptoCurrencyRepository.findAllById(cryptoCurrencyIds);
+
+        if (cryptoCurrencies.size() != cryptoCurrencyIds.size()){
+            throw new EntityNotFoundException("One or more cryptocurrencies not found.");
+        }
```

Comment:
Why this check is needed? Why can't we predict a scenario, when not all crypto currencies can be found and proceed with the next steps?